arbnode/seq_coordinator.go

```diff
@@ -584,13 +588,34 @@ func (c *SeqCoordinator) Start(ctxIn context.Context) {
 }
 
 func (c *SeqCoordinator) StopAndWait() {
+	wasChosen := c.CurrentlyChosen()
+	c.StopWaiter.StopAndWait()
+	// normal context now closed, use parent context
+	ctx := c.StopWaiter.GetParentContext()
 	if c.CurrentlyChosen() {
-		_ = c.chosenOneRelease(c.GetContext())
+		wasChosen = true
 	}
 	if c.reportedAlive {
-		_ = c.livelinessRelease(c.GetContext())
+		_ = c.livelinessRelease(ctx)
+	}
+	if wasChosen {
+		_ = c.chosenOneRelease(ctx)
```

Comment:
This needs to set `c.lockoutUntil` to 0, probably inside the `chosenOneRelease` function. I'd also log a warning if any of these calls fail